modules/contacts/server/controllers/contacts.server.controller.js

```diff
@@ -45,6 +48,29 @@ exports.add = function(req, res) {
         });
       }
 
+      // Check if contact already exists
+      Contact.findOne({
+        $or: [
+          { users: [ req.body.friendUserId, req.user._id ] },
+          { users: [ req.user._id, req.body.friendUserId ] }
+        ]
+      }).exec(function(err, existingContact) {
+        if (err) return done(err);
+
+        if (existingContact) {
+          // Contact already exists!
+          return res.status(400).json({
+            message: errorHandler.getErrorMessageByKey('invalid-id')
```

Comment:
You're right.

I'm not huge fan of passing error messages directly to front anyway though, because it easily ends up user seeing too technical messages. So I'd be inclined to see if the error can be reflected with correct 4xx status code. WDYT?